graphql/src/execution/execution.rs

```diff
@@ -894,59 +844,43 @@ pub fn coerce_variable_values(
     let mut coerced_values = HashMap::new();
     let mut errors = vec![];
 
-    if let Some(variable_definitions) = qast::get_variable_definitions(operation) {
-        for variable_def in variable_definitions.iter() {
-            // Skip variable if it has an invalid type
-            if !sast::is_input_type(&schema.document, &variable_def.var_type) {
-                errors.push(QueryExecutionError::InvalidVariableTypeError(
-                    variable_def.position,
-                    variable_def.name.to_owned(),
-                ));
-                continue;
-            }
-
-            let value = match variables {
-                None => None,
-                Some(vars) => vars.get(&variable_def.name),
-            };
+    for variable_def in qast::get_variable_definitions(operation)
+        .into_iter()
+        .flatten()
+    {
+        // Skip variable if it has an invalid type
+        if !sast::is_input_type(&schema.document, &variable_def.var_type) {
+            errors.push(QueryExecutionError::InvalidVariableTypeError(
+                variable_def.position,
+                variable_def.name.to_owned(),
+            ));
+            continue;
+        }
 
-            match value.map(|val| val.deref().to_owned()) {
-                // No variable value provided, either use the default or fail
-                None => {
-                    if let Some(ref default_value) = variable_def.default_value {
-                        coerced_values
-                            .insert(variable_def.name.to_owned(), default_value.to_owned());
-                    } else if let s::Type::NonNullType(_) = variable_def.var_type {
-                        errors.push(QueryExecutionError::MissingVariableError(
-                            variable_def.position,
-                            variable_def.name.to_owned(),
-                        ));
-                    }
-                }
+        let value = variables
+            .as_ref()
+            .and_then(|vars| vars.get(&variable_def.name));
 
-                // A null value was provided, either use it or fail
-                Some(q::Value::Null) => {
-                    if let s::Type::NonNullType(_) = variable_def.var_type {
-                        errors.push(QueryExecutionError::InvalidVariableError(
```

Comment:
I think this was the only place where we were using `InvalidVariableError`, so we can probably remove this enum variant now.

Reply:
Funny you commented on this because I also noticed that `InvalidVariableError` was dead, but didn't realize I made it so. The error message for this case got worse, and in general the coercion errors are not specific, but that's not something we want to solve now. I removed the dead variant.